sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,222 @@
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_first_session_ping,
+      CASE
+      WHEN
+        metrics.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_metrics_ping,
+      DATE(first_session.first_run_datetime) AS min_first_session_ping_run_date,
+      CASE
+        mozfun.norm.get_earliest_value(
+          [
+            (
+              STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)
+            ),
+            (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+          ]
+        )
+      WHEN
+        STRUCT(first_session.adjust_network, first_session.first_run_datetime)
+      THEN
+        'first_session'
+      WHEN
+        STRUCT(metrics.adjust_network, metrics.min_submission_datetime)
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS adjust_network__source_ping,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)),
+          (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+        ]
+      ).earliest_date AS adjust_network__source_ping_datetime,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        metrics.min_submission_datetime
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping_datetime
+    ) AS metadata
+  FROM
+    first_seen
+  FULL OUTER JOIN
+    first_session_ping first_session
+  USING
+    (client_id)
+  FULL OUTER JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id)
+),
+--existing clients in firefox_android_clients_v1
+_previous AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.fenix_derived.firefox_android_clients_v1`
+)
+SELECT
+  IF(_current.client_id IS NOT NULL, _current, _previous).* REPLACE (
+    COALESCE(_previous.device_manufacturer, _current.device_manufacturer) AS device_manufacturer,
+    COALESCE(_previous.device_model, _current.device_model) AS device_model,
+    COALESCE(_previous.os_version, _current.os_version) AS os_version,
+    COALESCE(_previous.adjust_campaign, _current.adjust_campaign) AS adjust_campaign,
+    COALESCE(_previous.adjust_ad_group, _current.adjust_ad_group) AS adjust_ad_group,
+    COALESCE(_previous.adjust_creative, _current.adjust_creative) AS adjust_creative,
+    COALESCE(_previous.adjust_network, _current.adjust_network) AS adjust_network,
+    COALESCE(_previous.install_source, _current.install_source) AS install_source,
+    STRUCT(
+      COALESCE(
+        _previous.metadata.reported_first_session_ping,
+        _current.metadata.reported_first_session_ping
+      ) AS reported_first_session_ping,
```

Comment:
These should be ORs, since they could have reported these pings previously or currently:
```suggestion
      _previous.metadata.reported_first_session_ping
        OR _current.metadata.reported_first_session_ping
        AS reported_first_session_ping,
```

Reply:
So if either previous or current reported (not null) it would be captured, it'd be equivalent right?